cli/js/files.ts

```diff
@@ -231,5 +281,82 @@ export type OpenMode =
  * @internal
  */
 export function create(filename: string): Promise<File> {
-  return open(filename, "w+");
+  return open(filename, {
+    read: true,
+    create: true,
+    write: true,
+    truncate: true
+  });
+}
+
+/** Check if OpenMode is set to something that is valid.
+ *  @returns Tuple representing if openMode is valid and error message if it's not
+ *  @internal
+ */
+function checkOpenMode(mode: OpenMode): [boolean, string] {
+  const allOptionsAreFalse =
+    Object.values(mode).filter(val => val == true).length === 0;
+  const truncateOptionWithoutWriteAccess = mode.truncate && !mode.write;
+  const createOrCreateNewWithoutWriteOrAppend =
+    (mode.create || mode.createNew) && !(mode.write || mode.append);
+  if (allOptionsAreFalse)
+    return [false, "OpenMode require at least one option to be true"];
+  if (truncateOptionWithoutWriteAccess)
+    return [false, "Truncate option require write accesss"];
+  if (createOrCreateNewWithoutWriteOrAppend)
+    return [false, "Create or create new requires write or append access"];
+  return [true, ""];
+}
+
+/** Converts OpenModeLegacy to Legacy
+ *  @internal
+ */
+function convertOpenModeLegacy(mode: OpenModeLegacy): OpenMode {
```

Comment:
Why is this legacy mode needed?

Reply:
I think we should keep it because the "w+" style arguments are very familiar to some. But maybe the name "legacy" isn't so good. OpenModePosix ? OpenModeAbbreviated ?